packages/node/test/integration/services/firebase-service.ts

```diff
@@ -36,7 +40,16 @@ class FirebaseMessagingService implements IMessagingService {
           );
           return;
         }
-        callback(snapshot.val());
+        const msg = snapshot.val();
+        const msgKey = JSON.stringify(msg);
+        if (msg !== null) {
```

Comment:
@cf19drofxots We can avoid if-nesting by flipping the condition and using `return`:

```ts
// Case 1: Message is null, don't do anything.
if (msg === null) {
  return;
}

// Case 2: Message is already served, delete it.
if (msgKey in this.servedMessages) {
  delete this.servedMessages[msgKey];
  return;
}

// Case 3: Message must be served, do it.
this.servedMessages[msgKey] = true;
callback(msg);
```

Reply:
Good call!